### ISSUE TITLE
Step results

### ISSUE THREAD

**Issue by pmagnuson** [theRealGetman/survey-flow#5]
Is there a way for a step to get the results of a previous step?

**Comment by thebedcoder** [comment]
Yes it is, but it's not so convenient. Unfortunately there is no direct way, but it's possible to use `onSubmit` callback.

1) On 1 step you can submit some result.
2) Handle it inside `onSubmit` and return list of next steps according to the results.

Tell me if that can fill your needs :)

**Comment by pmagnuson** [comment]
Here is the use case:

- step 1 - text input for 'what are you worried about?'
- step 2 - text input for 'where do you feel this in your body?'
- step 3 - go through an exercise (custom step)
- step 4 - are you still worried about 'result from step 1'?
- step 5 - how does your 'answer from step 2' feel now?

I can see how your suggestion would be feasible.  However, if results are needed from multiple steps, then there needs to be some kind of state tracking to decide which has just completed and which steps are next.  

One alternative is to use the _onPressed function of a CustomStepWidget.  However that then means that every surveyStep (where the results are needed) would need to be a CustomStep.  And results would have to be passed back up to a higher object in the widget tree. 

as a proposal, maybe an additional function in `SurveyFlow()`
```
onStepComplete: (List<StepResult> stepResults){}
```
This would be called at the completion of each step.  Then based on the last stepResult.stepId, decisions could be made about what information to save and use later.  Or when step4, 5 are reached, they could read the existing stepResults and use the information as needed.  

In contemplating this more, a mechanism as you suggested must be used.  Because the later steps are using dynamic information, the step cannot be defined ahead of time.  The later step must be created on the fly using previous information.  This also means that the step definitions cannot be defined ahead of time and downloaded from a server. 

The dart string syntax includes dynamically evaluated expressions - `'value of count is ${getTheCount()}'`.  If the value of stepResults was always available or as a function, then it might be possible to use these expressions to get the desired results.  And this would allow the steps to be predefined.  It would require each of the fromJson() parsers to evaluate the string expression.  

Hoping this makes sense.  Don't hesitate to point out something I've missed.  

Thanks!!
